### PR TITLE
Parse and save correctly low-penetrance pathogenic and likely pathogenic ClinVar info

### LINKED ISSUE #5048: Add "low_penetrance" to REV_CLINSIG_MAP
I got this warning while loading variants from a test case:

`WARNING Clinsig value low_penetrance does not have an internal representation`

image

Add to [this variable](https://github.com/Clinical-Genomics/scout/blob/f3a97783c658479c95025b34acf6452a7262a10d/scout/constants/clnsig.py#L19)

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #5048

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DNIL
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested by re-loading [the case](https://scout-stage.scilifelab.se/cust000/SWEDAC-2019-HG-05-Trio-NovaSeq) that triggered the warning

### Loading without warnings:

image


### Info saved in the database alright:

image

I just have to figure out why the info is not displayed on the variant page now.. 🤔

**Comment by northwestwitch** [comment]
Ah I'm an idiot, I wasn't thinking that in order to see a change in the web page I have to deploy it also on cg-vm1. Here it is:

image

This is the variant in question ---> https://scout-stage.scilifelab.se/cust000/SWEDAC-2019-HG-05-Trio-NovaSeq/39e574903c122fcd127000368be42330

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,9 +5,12 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 About changelog [here](https://keepachangelog.com/en/1.0.0/)
 
 ## [unreleased]
+### Added
+- Parse ClinVar low-penetrance info and display it alongside Pathogenic and likely pathogenic on SNVs pages
```
Comment (dnil) [inline CHANGELOG.md:9]: 🥇

File: scout/build/variant/clnsig.py [hunk 2]
```diff
@@ -24,4 +19,7 @@ def build_clnsig(clnsig_info):
         revstat=clnsig_info.get("revstat"),
     )
 
+    if isinstance(value, str) and "low_penetrance" in value:
+        clnsig_obj["low_penetrance"] = True
```
Comment (dnil) [inline scout/build/variant/clnsig.py:23]: 👍

File: scout/server/blueprints/variant/templates/variant/components.html [hunk 3]
```diff
@@ -17,7 +17,7 @@
       <tbody>
         {% for clinsig in variant.clinsig_human %}
           <tr>
-            <td>{{ clinsig.human }}</td>
+            <td>{{ clinsig.human }} {% if clinsig.low_penetrance %}(low penetrance){% endif %}</td>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/components.html:20]: Nice!

File: tests/build/test_build_clnsig.py [hunk 4]
```diff
@@ -39,3 +39,21 @@ def test_build_clnsig_pathogenic_int():
     assert isinstance(clsnig_obj, dict)
     ## THEN assert the value for pathogenic (5) is used
     assert clsnig_obj["value"] == 5
+
+
+def test_build_clnsig_low_penetrance():
```
Comment (dnil) [inline tests/build/test_build_clnsig.py:44]: 💯

File: scout/constants/clnsig.py [hunk 5]
```diff
@@ -33,7 +33,9 @@
     "conflicting_classifications_of_pathogenicity": 8,
     "uncertain_significance": 0,
     "likely_pathogenic": 4,
+    "likely_pathogenic,low_penetrance": 4,
```
Comment (dnil) [inline scout/constants/clnsig.py:36]: 👍 well spotted.

File: tests/parse/test_parse_clnsig.py [hunk 6]
```diff
@@ -91,6 +89,30 @@ def test_parse_modern_clnsig_clnvid(cyvcf2_variant):
     assert len(clnsig_annotations) == 2
 
 
+def test_parse_clnsig_low_penetrance(cyvcf2_variant):
+    """Testing parsing a variant with has a low penetrance clnsig (pathogenic or likely pathogenic)"""
+
+    ## GIVEN a variant with the following ClinVar annotations
+    acc_nr = "7888"
+    clnsig = "Pathogenic/Likely_pathogenic/Pathogenic&_low_penetrance"
```
Comment (dnil) [inline tests/parse/test_parse_clnsig.py:97]: 👍

File: tests/parse/test_parse_clnsig.py [hunk 7]
```diff
@@ -91,6 +89,30 @@ def test_parse_modern_clnsig_clnvid(cyvcf2_variant):
...
+
+    cyvcf2_variant.INFO["CLNVID"] = acc_nr
+    cyvcf2_variant.INFO["CLNSIG"] = clnsig
+    cyvcf2_variant.INFO["CLNREVSTAT"] = revstat
+
+    ## WHEN parsing the annotations
+    clnsig_annotations = parse_clnsig(cyvcf2_variant)
+
+    ## THEN assert that clnsig is parsed correctly
+    assert set(["pathogenic", "likely_pathogenic", "pathogenic,low_penetrance"]) == {
```
Comment (dnil) [inline tests/parse/test_parse_clnsig.py:108]: 👍

**Review by dnil — APPROVED** [review]
Looks good to me!

What does this mean for previous cases? We have represented all "low_penetrance" as "other"? They would still have been loaded, due to the keyword "pathogenic" in the string, right? And is this new from this year, or have we had it longer?

**Comment by dnil** [comment]
> Looks good to me!
> 
> What does this mean for previous cases? We have represented all "low_penetrance" as "other"? They would still have been loaded, due to the keyword "pathogenic" in the string, right? And is this new from this year, or have we had it longer?

Ah, no, its rather that the "low penetrance" was just not represented at all, only with the main term, ie "Pathogenic" for a "Pathogenic, low penetrance"?

**Comment by northwestwitch** [comment]
> Ah, no, its rather that the "low penetrance" was just not represented at all, only with the main term, ie "Pathogenic" for a "Pathogenic, low penetrance"?

Precisely. When you had "Pathogenic, low penetrance", then the parsing of the second part after the comma failed, generating the warning when the variants were loaded. Only "Pathogenic" was saved under clnsig of the variant, which is correct but also incomplete info I think.

If you check the [ClinVar pages](https://www.ncbi.nlm.nih.gov/clinvar/docs/clinsig/#standard), pathogenic LP and likely pathogenic LP are categories apart:

image

So it would make sense to assign a specific category to these 2 as well. The biggest problem is that there is no corresponding int to these categories, like you have for pathogenic (5) or likely pathogenic (4). Of course you could create one but it would be arbitrary and disconnected from ClinVar, where the ints were originally taken from.

I feel this way is safe because we are not altering what is saved into the database for the variant (4 and 5), but we just add more info.

**Comment by northwestwitch** [comment]
I could make this PR a patch perhaps, so we deploy it with the rest?

**Pull request merged**